var/spack/repos/builtin/packages/py-cython/package.py

```diff
@@ -41,6 +42,14 @@ class PyCython(PythonPackage):
 
     depends_on("python@2.7:2,3.4:", when="@3:", type=("build", "link", "run"))
     depends_on("python@2.6:2,3.3:", type=("build", "link", "run"))
+
+    # https://github.com/cython/cython/commit/1cd24026e9cf6d63d539b359f8ba5155fd48ae21
+    # collections.Iterable was removed in Python 3.10
+    depends_on("python@:3.9", when="@:0.29.14", type=("build", "link", "run"))
+
+    # https://github.com/cython/cython/commit/430e2ca220c8fed49604daf578df98aadb33a87d
+    depends_on("python@:3.8", when="@:0.29.13", type=("build", "link", "run"))
+
```

Comment:
Any idea what the first version that added support for Python 3.8 (or 3.11) was? That would help with #5307 because it would allow us to deprecate and remove old versions that aren't patched and don't work anyway.